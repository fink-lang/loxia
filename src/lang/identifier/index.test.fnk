{fink2js} = import '../../testing'
{describe, it, expect, to_match_snapshot} = import '@fink/jest'


describe 'identifiers', fn:
  it 'escapes reserved JS identifiers', fn:
    expect
      fink2js `
        {do, if, in, for, let, new, try, var, case, enum, void} = foobar
        {with, break, catch, class, const, super, while, yield, delete} = spam
        {export, import, public, return, static, switch, typeof, default} = ham
        {extends, finally, package, private, continue, function} = ni
        {arguments, interface, protected, implements, instanceof} = nu
      `
      to_match_snapshot


  it 'does not escape true, false, null', fn:
    expect
      fink2js `
        foo = [true, false, null]
      `
      to_match_snapshot


  it 'escapes identifiers incompatible with JS', fn:
    expect
      fink2js `
        @ = spam
        ∂ = shrub
        ∑ = ni
      `
      to_match_snapshot


