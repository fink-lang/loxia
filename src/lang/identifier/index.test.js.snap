// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identifiers does not escape true, false, null 1`] = `
"const foo = [true, false, null];
Object.assign(module.exports, {
  foo
});"
`;

exports[`identifiers escapes identifiers incompatible with JS 1`] = `
"const ˆ1s = spam;
const ˆ6pu = shrub;
Object.assign(module.exports, {
  ˆ1s,
  ˆ6pu
});"
`;

exports[`identifiers escapes reserved JS identifiers 1`] = `
"const {
  \\"do\\": ˆdo,
  \\"if\\": ˆif,
  \\"in\\": ˆin,
  \\"for\\": ˆfor,
  \\"let\\": ˆlet,
  \\"new\\": ˆnew,
  \\"try\\": ˆtry,
  \\"var\\": ˆvar,
  \\"case\\": ˆcase,
  \\"enum\\": ˆenum,
  \\"void\\": ˆvoid
} = foobar;
const {
  \\"with\\": ˆwith,
  \\"break\\": ˆbreak,
  \\"catch\\": ˆcatch,
  \\"class\\": ˆclass,
  \\"const\\": ˆconst,
  \\"super\\": ˆsuper,
  \\"while\\": ˆwhile,
  \\"yield\\": ˆyield,
  \\"delete\\": ˆdelete
} = spam;
const {
  \\"export\\": ˆexport,
  \\"import\\": ˆimport,
  \\"public\\": ˆpublic,
  \\"return\\": ˆreturn,
  \\"static\\": ˆstatic,
  \\"switch\\": ˆswitch,
  \\"typeof\\": ˆtypeof,
  \\"default\\": ˆdefault
} = ham;
const {
  \\"extends\\": ˆextends,
  \\"finally\\": ˆfinally,
  \\"package\\": ˆpackage,
  \\"private\\": ˆprivate,
  \\"continue\\": ˆcontinue,
  \\"debugger\\": ˆdebugger,
  \\"function\\": ˆfunction
} = ni;
const {
  \\"arguments\\": ˆarguments,
  \\"interface\\": ˆinterface,
  \\"protected\\": ˆprotected,
  \\"implements\\": ˆimplements,
  \\"instanceof\\": ˆinstanceof
} = nu;
Object.assign(module.exports, {});"
`;
